diff --git a/node_modules/@expo/cli/build/src/start/interface/interactiveActions.js b/node_modules/@expo/cli/build/src/start/interface/interactiveActions.js
index 7941956..a40663e 100644
--- a/node_modules/@expo/cli/build/src/start/interface/interactiveActions.js
+++ b/node_modules/@expo/cli/build/src/start/interface/interactiveActions.js
@@ -78,12 +78,17 @@ class DevServerManagerActions {
         if (this.devServerManager.getNativeDevServerPort()) {
             const devServer = this.devServerManager.getDefaultDevServer();
             try {
+                const devServerUrl = devServer.getDevServerUrl();
                 const nativeRuntimeUrl = devServer.getNativeRuntimeUrl();
                 const interstitialPageUrl = devServer.getRedirectUrl();
                 (0, _commandsTable.printQRCode)(interstitialPageUrl ?? nativeRuntimeUrl);
                 if (interstitialPageUrl) {
                     _log.log((0, _commandsTable.printItem)((0, _chalk().default)`Choose an app to open your project at {underline ${interstitialPageUrl}}`));
                 }
+                if (nativeRuntimeUrl !== devServerUrl) {
+                  // When using a custom scheme, the IP in the deep link URL is difficult to read, so we print the LAN address too.
+                  Log.log((0, _commandsTable).printItem(_chalk.default`Dev server LAN address: {underline ${devServerUrl}}`));
+                }
                 if (_env.env.__EXPO_E2E_TEST) {
                     // Print the URL to stdout for tests
                     console.info(`[__EXPO_E2E_TEST:server] ${JSON.stringify({
diff --git a/node_modules/@expo/cli/build/src/start/server/BundlerDevServer.js b/node_modules/@expo/cli/build/src/start/server/BundlerDevServer.js
index 086c102..90b9952 100644
--- a/node_modules/@expo/cli/build/src/start/server/BundlerDevServer.js
+++ b/node_modules/@expo/cli/build/src/start/server/BundlerDevServer.js
@@ -291,6 +291,16 @@ class BundlerDevServer {
         if (options.hostType === "localhost") {
             return `${location.protocol}://localhost:${location.port}`;
         }
+        if (location.host === 'localhost') {
+          // try autodetect IP
+          try {
+            const ip = require("../../utils/ip").getIpAddress();
+            if (ip !== "127.0.0.1") {
+              return `${location.protocol}://${ip}:${location.port}`;
+            }
+          } catch(e) {}
+            // if the above fails, fall back to localhost
+        }
         return location.url ?? null;
     }
     getDevServerUrlOrAssert(options = {}) {
diff --git a/node_modules/@expo/cli/build/src/utils/scheme.js b/node_modules/@expo/cli/build/src/utils/scheme.js
index 476fc9f..d3682c6 100644
--- a/node_modules/@expo/cli/build/src/utils/scheme.js
+++ b/node_modules/@expo/cli/build/src/utils/scheme.js
@@ -125,7 +125,9 @@ function sortLongest(obj) {
 }
 async function getSchemesForIosAsync(projectRoot) {
     try {
-        const infoPlistBuildProperty = (0, _getInfoPlistPath().getInfoPlistPathFromPbxproj)(projectRoot);
+        // see https://github.com/expo/expo/discussions/20875 for details on this patch
+        // if this patch ever breaks, it can be worked around by adding a --scheme flag to the expo start command
+        const infoPlistBuildProperty = (0, _getInfoPlistPath).getInfoPlistPathFromPbxproj(projectRoot, { buildConfiguration: 'Debug' });
         debug(`ios application Info.plist path:`, infoPlistBuildProperty);
         if (infoPlistBuildProperty) {
             const configPath = _path().default.join(projectRoot, "ios", infoPlistBuildProperty);
@@ -146,6 +148,14 @@ async function getSchemesForAndroidAsync(projectRoot) {
         const configPath = await _configPlugins().AndroidConfig.Paths.getAndroidManifestAsync(projectRoot);
         const manifest = await _configPlugins().AndroidConfig.Manifest.readAndroidManifestAsync(configPath);
         const schemes = await _configPlugins().AndroidConfig.Scheme.getSchemesFromManifest(manifest);
+        // see https://github.com/expo/expo/discussions/20875 for details on this patch
+        // if this patch ever breaks, it can be worked around by adding a --scheme flag to the expo start command
+        if (schemes.length < 1) {
+          const debugManifestPath = _path.default.join(projectRoot, 'android/app/src/debug/AndroidManifest.xml');
+          const debugManifest = await _configPlugins.AndroidConfig.Manifest.readAndroidManifestAsync(debugManifestPath);
+          const debugSchemes = await _configPlugins.AndroidConfig.Scheme.getSchemesFromManifest(debugManifest);
+          schemes.push(...debugSchemes);
+        }
         debug(`android application schemes:`, schemes);
         return resolveExpoOrLongestScheme(schemes);
     } catch (error) {
