diff --git a/node_modules/@rainbow-me/rainbowkit/dist/chunk-R7UVC5N6.js b/node_modules/@rainbow-me/rainbowkit/dist/chunk-R7UVC5N6.js
index 275c804..eab512c 100644
--- a/node_modules/@rainbow-me/rainbowkit/dist/chunk-R7UVC5N6.js
+++ b/node_modules/@rainbow-me/rainbowkit/dist/chunk-R7UVC5N6.js
@@ -1625,8 +1625,8 @@ function abbreviateETHBalance(number) {
 
 // src/components/ConnectButton/formatAddress.ts
 function formatAddress(address) {
-  const leadingChars = 4;
-  const trailingChars = 4;
+  const leadingChars = 8;
+  const trailingChars = 10;
   return address.length < leadingChars + trailingChars ? address : `${address.substring(0, leadingChars)}\u2026${address.substring(address.length - trailingChars)}`;
 }
 
@@ -4590,5 +4590,6 @@ export {
   useAccountModal,
   useChainModal,
   useConnectModal,
-  ConnectButton
+  ConnectButton,
+  useModalState,
 };
diff --git a/node_modules/@rainbow-me/rainbowkit/dist/index.js b/node_modules/@rainbow-me/rainbowkit/dist/index.js
index 8630ea1..a6fac1c 100644
--- a/node_modules/@rainbow-me/rainbowkit/dist/index.js
+++ b/node_modules/@rainbow-me/rainbowkit/dist/index.js
@@ -22,7 +22,8 @@ import {
   useChainId,
   useChainModal,
   useConnectModal,
-  useTransactionStore
+  useTransactionStore,
+  useModalState,
 } from "./chunk-R7UVC5N6.js";
 import {
   lightTheme
@@ -701,5 +702,6 @@ export {
   useAddRecentTransaction,
   useChainModal,
   useConnectModal,
-  wallet
+  wallet,
+  useModalState,
 };
