diff --git a/node_modules/@react-native-community/cameraroll/ios/.RNCCameraRollManager.m.swp b/node_modules/@react-native-community/cameraroll/ios/.RNCCameraRollManager.m.swp
new file mode 100644
index 0000000..5f05a91
Binary files /dev/null and b/node_modules/@react-native-community/cameraroll/ios/.RNCCameraRollManager.m.swp differ
diff --git a/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m b/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
index 566d8fc..862aa42 100644
--- a/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
+++ b/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
@@ -236,7 +236,7 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
   
   // Predicate for fetching assets within a collection
   PHFetchOptions *const assetFetchOptions = [RCTConvert PHFetchOptionsFromMediaType:mediaType];
-  assetFetchOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:NO]];
+  assetFetchOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"modificationDate" ascending:NO]];
   
   BOOL __block foundAfter = NO;
   BOOL __block hasNextPage = NO;
