diff --git a/node_modules/@react-native-community/cameraroll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java b/node_modules/@react-native-community/cameraroll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
index 63896c8..6e96331 100644
--- a/node_modules/@react-native-community/cameraroll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
+++ b/node_modules/@react-native-community/cameraroll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
@@ -13,6 +13,7 @@ import android.content.Context;
 import android.content.res.AssetFileDescriptor;
 import android.database.Cursor;
 import android.graphics.BitmapFactory;
+import android.media.ExifInterface;
 import android.media.MediaMetadataRetriever;
 import android.media.MediaScannerConnection;
 import android.net.Uri;
@@ -521,13 +522,21 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
         retriever.setDataSource(photoDescriptor.getFileDescriptor());
 
         try {
-          if (width <= 0 || height <= 0) {
-            width =
-                Integer.parseInt(
-                    retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH));
-            height =
-                Integer.parseInt(
-                    retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT));
+          int videoWidth = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH)
+          );
+          int videoHeight = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT)
+          );
+          int orientation = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION)
+          );
+          if (Math.abs(orientation) % 180 == 90) {
+            width = videoHeight;
+            height = videoWidth;
+          } else {
+            width = videoWidth;
+            height = videoHeight;
           }
           int timeInMillisec =
               Integer.parseInt(
@@ -549,6 +558,24 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
         FLog.e(ReactConstants.TAG, "Could not get video metadata for " + photoUri.toString(), e);
         return false;
       }
+    } else {
+      try {
+        ExifInterface exif = new ExifInterface(photoUri.toString().substring(7));
+        int orientation = exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL);
+        if (
+          orientation == ExifInterface.ORIENTATION_ROTATE_90 ||
+          orientation == ExifInterface.ORIENTATION_ROTATE_270 ||
+          orientation == ExifInterface.ORIENTATION_TRANSPOSE ||
+          orientation == ExifInterface.ORIENTATION_TRANSVERSE
+        ) {
+          float oldHeight = height;
+          height = width;
+          width = oldHeight;
+        }
+      } catch (IOException e) {
+        FLog.e(ReactConstants.TAG, "Could not get EXIF info for " + photoUri.toString(), e);
+        return false;
+      }
     }
 
     if (width <= 0 || height <= 0) {
diff --git a/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m b/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
index 7170483..27158c9 100644
--- a/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
+++ b/node_modules/@react-native-community/cameraroll/ios/RNCCameraRollManager.m
@@ -264,7 +264,7 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
   
   // Predicate for fetching assets within a collection
   PHFetchOptions *const assetFetchOptions = [RCTConvert PHFetchOptionsFromMediaType:mediaType fromTime:fromTime toTime:toTime];
-  assetFetchOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"creationDate" ascending:NO]];
+  assetFetchOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"modificationDate" ascending:NO]];
   
   BOOL __block foundAfter = NO;
   BOOL __block hasNextPage = NO;
