diff --git a/node_modules/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt b/node_modules/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt
index 6496976..1e3550e 100644
--- a/node_modules/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt
+++ b/node_modules/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt
@@ -30,6 +30,7 @@ class SplashScreenController(
 
   fun showSplashScreen(successCallback: () -> Unit = {}) {
     weakActivity.get()?.runOnUiThread {
+      (splashScreenView.parent as? ViewGroup)?.removeView(splashScreenView)
       contentView.addView(splashScreenView)
       splashScreenShown = true
       successCallback()
