diff --git a/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle b/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
index ae5f09c..ee4a937 100644
--- a/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
+++ b/node_modules/expo-modules-autolinking/scripts/android/autolinking_implementation.gradle
@@ -159,7 +159,7 @@ if (rootProject instanceof ProjectDescriptor) {
   }
 
   def addDependencies = { DependencyHandler handler, Project project ->
-    ExpoAutolinkingManager manager = gradle.ext.expoAutolinkingManager
+    def manager = gradle.ext.expoAutolinkingManager
     def modules = manager.getModules()
 
     if (!modules.length) {
@@ -193,7 +193,7 @@ if (rootProject instanceof ProjectDescriptor) {
       return
     }
 
-    ExpoAutolinkingManager manager = gradle.ext.expoAutolinkingManager
+    def manager = gradle.ext.expoAutolinkingManager
 
     if (rootProject.findProject(':expo-modules-core')) {
       // `expo` requires `expo-modules-core` as a dependency, even if autolinking is turned off.
