diff --git a/node_modules/expo-media-library/ios/EXMediaLibrary.podspec b/node_modules/expo-media-library/ios/EXMediaLibrary.podspec
index 7cc9e70..a5b5336 100644
--- a/node_modules/expo-media-library/ios/EXMediaLibrary.podspec
+++ b/node_modules/expo-media-library/ios/EXMediaLibrary.podspec
@@ -18,4 +18,5 @@ Pod::Spec.new do |s|
 
   s.dependency 'UMCore'
   s.dependency 'UMPermissionsInterface'
+  s.dependency 'UMFileSystemInterface'
 end
