diff --git a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
index 78e6fa4..6ff078b 100644
--- a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
+++ b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
@@ -20,7 +20,7 @@ @implementation Orientation
 }
 
 #if (!TARGET_OS_TV)
-static UIInterfaceOrientationMask _orientation = UIInterfaceOrientationMaskAll;
+static UIInterfaceOrientationMask _orientation = UIInterfaceOrientationMaskPortrait;
 
 + (void)setOrientation: (UIInterfaceOrientationMask)orientation {
     
