diff --git a/node_modules/react-native-swipe-gestures/index.js b/node_modules/react-native-swipe-gestures/index.js
index caa2a53..6737084 100644
--- a/node_modules/react-native-swipe-gestures/index.js
+++ b/node_modules/react-native-swipe-gestures/index.js
@@ -24,13 +24,7 @@ class GestureRecognizer extends Component {
   constructor(props, context) {
     super(props, context);
     this.swipeConfig = Object.assign(swipeConfig, props.config);
-  }
-
-  componentWillReceiveProps(props) {
-    this.swipeConfig = Object.assign(swipeConfig, props.config);
-  }
 
-  componentWillMount() {
     const responderEnd = this._handlePanResponderEnd.bind(this);
     const shouldSetResponder = this._handleShouldSetPanResponder.bind(this);
     this._panResponder = PanResponder.create({ //stop JS beautify collapse
@@ -41,6 +35,12 @@ class GestureRecognizer extends Component {
     });
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.config !== prevProps.config) {
+      this.swipeConfig = Object.assign(swipeConfig, this.props.config);
+    }
+  }
+
   _handleShouldSetPanResponder(evt, gestureState) {
     return evt.nativeEvent.touches.length === 1 && !this._gestureIsClick(gestureState);
   }
