diff --git a/node_modules/react-native-vector-icons/dist/index.js.flow b/node_modules/react-native-vector-icons/dist/index.js.flow
index 75c5d2d..4481676 100644
--- a/node_modules/react-native-vector-icons/dist/index.js.flow
+++ b/node_modules/react-native-vector-icons/dist/index.js.flow
@@ -12,6 +12,7 @@ export type IconButtonProps<Glyphs: string> = {
   color?: Color,
   name: Glyphs,
   size?: number,
+  ...
 };
 
 declare class IconButton<Glyphs: string> extends PureComponent<
@@ -58,6 +59,7 @@ export type IconProps<Glyphs: string> = {
   color?: Color,
   name: Glyphs,
   size?: number,
+  ...
 };
 
 export type ImageSource = {|
diff --git a/node_modules/react-native-vector-icons/index.js.flow b/node_modules/react-native-vector-icons/index.js.flow
index 75c5d2d..4481676 100644
--- a/node_modules/react-native-vector-icons/index.js.flow
+++ b/node_modules/react-native-vector-icons/index.js.flow
@@ -12,6 +12,7 @@ export type IconButtonProps<Glyphs: string> = {
   color?: Color,
   name: Glyphs,
   size?: number,
+  ...
 };
 
 declare class IconButton<Glyphs: string> extends PureComponent<
@@ -58,6 +59,7 @@ export type IconProps<Glyphs: string> = {
   color?: Color,
   name: Glyphs,
   size?: number,
+  ...
 };
 
 export type ImageSource = {|
