diff --git a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
index 3c8d37e..069e216 100644
--- a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
+++ b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
@@ -13,7 +13,6 @@ public class CustomKeyboardRootViewShadow extends LayoutShadowNode {
         mLayout.setShadowNode(this);
     }
 
-    @Override
     public void onBeforeLayout() {
         mLayout.setShadowNode(this);
     }
diff --git a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
index 2a31b47..21090bf 100644
--- a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
+++ b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
@@ -222,7 +222,7 @@ -(UIView*)getFirstResponder:(UIView*)view
     }
 }
 
--(void)changeKeyboardHeightForInput:(nonnull NSNumber*)inputFieldTag newHeight:(CGFloat)newHeight
+RCT_EXPORT_METHOD(changeKeyboardHeightForInput:(nonnull NSNumber*)inputFieldTag newHeight:(CGFloat)newHeight)
 {
     UIView* inputField = [self.bridge.uiManager viewForReactTag:inputFieldTag];
     if(inputField != nil)
diff --git a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomKeyboardViewController.m b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomKeyboardViewController.m
index e856264..cc5ba15 100644
--- a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomKeyboardViewController.m
+++ b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomKeyboardViewController.m
@@ -7,8 +7,8 @@
 
 #import "RCTCustomKeyboardViewController.h"
 
-#if __has_include(<KeyboardTrackingView/ObservingInputAccessoryView.h>)
-    #import <KeyboardTrackingView/ObservingInputAccessoryView.h>
+#if __has_include(<react-native-keyboard-tracking-view/ObservingInputAccessoryView.h>)
+    #import <react-native-keyboard-tracking-view/ObservingInputAccessoryView.h>
     #define ObservingInputAccessoryView_IsAvailable true
 #endif
 
diff --git a/node_modules/react-native-keyboard-input/react-native-keyboard-input.podspec b/node_modules/react-native-keyboard-input/react-native-keyboard-input.podspec
new file mode 100644
index 0000000..9f724be
--- /dev/null
+++ b/node_modules/react-native-keyboard-input/react-native-keyboard-input.podspec
@@ -0,0 +1,18 @@
+require 'json'
+version = JSON.parse(File.read('package.json'))["version"]
+
+Pod::Spec.new do |s|
+
+  s.name                  = "react-native-keyboard-input"
+  s.version               = version
+  s.summary               = "Keyboard Input for React Native."
+  s.homepage              = "https://github.com/wix/react-native-keyboard-input"
+  s.license               = 'MIT'
+  s.source                = { path: '.' }
+  s.ios.deployment_target = '9.0'
+  s.authors               = { 'Leo Natan' => 'lnatan@wix.com' }
+  s.source_files          = 'lib/ios/{LNInterpolation/*,RCTCustomInputController/*}.{h,m,mm}'
+  s.dependency 'React'
+  s.dependency 'react-native-keyboard-tracking-view'
+
+end 
diff --git a/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js b/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
index 20d61c9..13a6493 100644
--- a/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
+++ b/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
@@ -41,6 +41,12 @@ export default class TextInputKeyboardManagerIOS {
       }
     }
   };
+
+  static setKeyboardHeight(textInputRef, keyboardHeight) {
+    const reactTag = findNodeHandle(textInputRef);
+    CustomInputController.changeKeyboardHeightForInput(reactTag, keyboardHeight);
+  }
+
 }
 
 function findNodeHandle(ref) {
