diff --git a/node_modules/react-native-keyboard-input/lib/android/build.gradle b/node_modules/react-native-keyboard-input/lib/android/build.gradle
index 1dd0e57..ec8db1e 100644
--- a/node_modules/react-native-keyboard-input/lib/android/build.gradle
+++ b/node_modules/react-native-keyboard-input/lib/android/build.gradle
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 34
     buildToolsVersion "28.0.3"
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 28
+        targetSdkVersion 34
         versionCode 1
         versionName "1.0"
     }
diff --git a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
index 46d2207..830c7b5 100644
--- a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
+++ b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/CustomKeyboardRootViewShadow.java
@@ -14,7 +14,6 @@ public class CustomKeyboardRootViewShadow extends LayoutShadowNode {
         mLayout.setShadowNode(this);
     }
 
-    @Override
     public void onBeforeLayout(NativeViewHierarchyOptimizer nativeViewHierarchyOptimizer) {
         mLayout.setShadowNode(this);
     }
diff --git a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/ReactSoftKeyboardMonitor.java b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/ReactSoftKeyboardMonitor.java
index 6bd4453..a0f588a 100644
--- a/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/ReactSoftKeyboardMonitor.java
+++ b/node_modules/react-native-keyboard-input/lib/android/src/main/java/com/wix/reactnativekeyboardinput/ReactSoftKeyboardMonitor.java
@@ -1,6 +1,11 @@
 package com.wix.reactnativekeyboardinput;
 
+import android.graphics.Point;
 import android.graphics.Rect;
+import android.os.Build;
+import android.util.Log;
+import android.util.TypedValue;
+import android.view.Display;
 import android.view.ViewTreeObserver;
 import android.view.Window;
 
@@ -65,8 +70,33 @@ public class ReactSoftKeyboardMonitor implements ReactScreenMonitor.Listener {
     private Listener mExternalListener;
     private ReactRootView mLastReactRootView;
 
+    private int translucentNavBarHeight;
+
     public ReactSoftKeyboardMonitor(ReactScreenMonitor screenMonitor) {
         screenMonitor.addListener(this);
+
+        this.translucentNavBarHeight = 0;
+
+        if (getWindow() == null) {
+            return;
+        }
+
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+
+        TypedValue typedValue = new TypedValue();
+        getWindow().getContext().getTheme().resolveAttribute(android.R.attr.windowTranslucentNavigation, typedValue, true);
+        if (typedValue.data == 0) {
+            return;
+        }
+
+        Display defaultDisplay = getWindow().getWindowManager().getDefaultDisplay();
+        Point size = new Point();
+        defaultDisplay.getSize(size);
+        Point realSize = new Point();
+        defaultDisplay.getRealSize(realSize);
+        this.translucentNavBarHeight = realSize.y - size.y;
     }
 
     @Override
@@ -132,21 +162,21 @@ public class ReactSoftKeyboardMonitor implements ReactScreenMonitor.Listener {
         RuntimeUtils.runOnUIThread(new Runnable() {
             @Override
             public void run() {
-                final Integer locallyVisibleHeight = getLocallyVisibleHeight();
-                if (locallyVisibleHeight == null) {
+                final Integer viewportVisibleHeight = getViewportVisibleHeight();
+                if (viewportVisibleHeight == null) {
                     // Too late to join the party - react-view seems to be gone...
                     return;
                 }
 
                 if (mLocallyVisibleHeight == null) {
-                    mLocallyVisibleHeight = locallyVisibleHeight;
+                    mLocallyVisibleHeight = viewportVisibleHeight;
                     mKeyboardHeight = mLocallyVisibleHeight;
                     Logger.d(TAG, "mLocallyVisibleHeight WAS NULL, now is: " + mLocallyVisibleHeight);
-                } else if (mLocallyVisibleHeight > locallyVisibleHeight) {
-                    mKeyboardHeight = mLocallyVisibleHeight - locallyVisibleHeight;
+                } else if (mMaxViewportVisibleHeight > viewportVisibleHeight) {
+                    mKeyboardHeight = mMaxViewportVisibleHeight - viewportVisibleHeight + translucentNavBarHeight;
                 } else {
-                    mKeyboardHeight = locallyVisibleHeight;
-                    Logger.d(TAG, "mKeyboardHeight = " + mKeyboardHeight + " mLocallyVisibleHeight = " + mLocallyVisibleHeight + " locallyVisibleHeight = " + locallyVisibleHeight);
+                    mKeyboardHeight = viewportVisibleHeight;
+                    Logger.d(TAG, "mKeyboardHeight = " + mKeyboardHeight + " mLocallyVisibleHeight = " + mLocallyVisibleHeight + " viewportVisibleHeight= " + viewportVisibleHeight);
                 }
             }
         });
diff --git a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
index 2a31b47..21090bf 100644
--- a/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
+++ b/node_modules/react-native-keyboard-input/lib/ios/RCTCustomInputController/RCTCustomInputController.m
@@ -222,7 +222,7 @@ -(UIView*)getFirstResponder:(UIView*)view
     }
 }
 
--(void)changeKeyboardHeightForInput:(nonnull NSNumber*)inputFieldTag newHeight:(CGFloat)newHeight
+RCT_EXPORT_METHOD(changeKeyboardHeightForInput:(nonnull NSNumber*)inputFieldTag newHeight:(CGFloat)newHeight)
 {
     UIView* inputField = [self.bridge.uiManager viewForReactTag:inputFieldTag];
     if(inputField != nil)
diff --git a/node_modules/react-native-keyboard-input/src/CustomKeyboardView.js b/node_modules/react-native-keyboard-input/src/CustomKeyboardView.js
index 59cae03..ec6f756 100644
--- a/node_modules/react-native-keyboard-input/src/CustomKeyboardView.js
+++ b/node_modules/react-native-keyboard-input/src/CustomKeyboardView.js
@@ -75,16 +75,16 @@ export default class CustomKeyboardView extends Component {
     }
   }
 
-  async UNSAFE_componentWillReceiveProps(nextProps) { //eslint-disable-line
-    const {inputRef, component, initialProps, onRequestShowKeyboard} = nextProps;
+  async componentDidUpdate(prevProps) {
+    const {inputRef, component, initialProps, onRequestShowKeyboard} = this.props;
 
     if (IsAndroid) {
-      if (this.props.component !== component && !component) {
+      if (prevProps.component !== component && !component) {
         await TextInputKeyboardManagerAndroid.reset();
       }
     }
 
-    if (IsIOS && TextInputKeyboardManagerIOS && inputRef && component !== this.props.component) {
+    if (IsIOS && TextInputKeyboardManagerIOS && inputRef && component !== prevProps.component) {
       if (component) {
         TextInputKeyboardManagerIOS.setInputComponent(inputRef, {component, initialProps});
       } else {
@@ -98,7 +98,7 @@ export default class CustomKeyboardView extends Component {
         onRequestShowKeyboard(args.keyboardId);
       });
     }
-    this.registerListener(this.props, nextProps);
+    this.registerListener(prevProps, this.props);
   }
 
   registerListener(props, nextProps) {
diff --git a/node_modules/react-native-keyboard-input/src/KeyboardAccessoryView.js b/node_modules/react-native-keyboard-input/src/KeyboardAccessoryView.js
index b91aa83..f403d38 100644
--- a/node_modules/react-native-keyboard-input/src/KeyboardAccessoryView.js
+++ b/node_modules/react-native-keyboard-input/src/KeyboardAccessoryView.js
@@ -126,11 +126,15 @@ export default class KeyboardAccessoryView extends Component {
   }
 
   render() {
+    let onLayout = undefined;
+    if (this.props.onHeightChanged) {
+      onLayout = this.onContainerComponentHeightChanged;
+    }
     return (
       <KeyboardTrackingView
         ref={r => this.trackingViewRef = r}
         style={styles.trackingToolbarContainer}
-        onLayout={this.onContainerComponentHeightChanged}
+        onLayout={onLayout}
         scrollBehavior={this.getIOSTrackingScrollBehavior()}
         revealKeyboardInteractive={this.props.revealKeyboardInteractive}
         manageScrollView={this.props.manageScrollView}
diff --git a/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js b/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
index 20d61c9..e8369f0 100644
--- a/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
+++ b/node_modules/react-native-keyboard-input/src/TextInputKeyboardMangerIOS.js
@@ -41,6 +41,11 @@ export default class TextInputKeyboardManagerIOS {
       }
     }
   };
+
+  static setKeyboardHeight(textInputRef, keyboardHeight) {
+    const reactTag = findNodeHandle(textInputRef);
+    CustomInputController.changeKeyboardHeightForInput(reactTag, keyboardHeight);
+  }
 }
 
 function findNodeHandle(ref) {
