diff --git a/node_modules/react-native-hyperlink/dist/index.js b/node_modules/react-native-hyperlink/dist/index.js
index a63c4ee..a345761 100644
--- a/node_modules/react-native-hyperlink/dist/index.js
+++ b/node_modules/react-native-hyperlink/dist/index.js
@@ -52,13 +52,11 @@ var Hyperlink = function (_Component) {
   }
 
   _createClass(Hyperlink, [{
-    key: 'componentWillReceiveProps',
-    value: function componentWillReceiveProps() {
-      var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
-          _ref$linkify = _ref.linkify,
-          linkify = _ref$linkify === undefined ? require('linkify-it')() : _ref$linkify;
-
-      this.linkifyIt = linkify;
+    key: 'componentDidUpdate',
+    value: function componentDidUpdate(prevProps) {
+      if (this.props.linkify !== prevProps.linkify) {
+        this.linkifyIt = this.props.linkify || require('linkify-it')()
+      }
     }
   }, {
     key: 'render',
diff --git a/node_modules/react-native-hyperlink/src/Hyperlink.js b/node_modules/react-native-hyperlink/src/Hyperlink.js
index ad90132..9406200 100644
--- a/node_modules/react-native-hyperlink/src/Hyperlink.js
+++ b/node_modules/react-native-hyperlink/src/Hyperlink.js
@@ -23,8 +23,10 @@ class Hyperlink extends Component {
     this.linkifyIt = props.linkify || require('linkify-it')()
   }
 
-  componentWillReceiveProps ({ linkify = require('linkify-it')() } = {}) {
-    this.linkifyIt = linkify
+  componentDidUpdate(prevProps) {
+    if (this.props.linkify !== prevProps.linkify) {
+      this.linkifyIt = this.props.linkify || require('linkify-it')()
+    }
   }
 
   render() {
