diff --git a/node_modules/react-native-background-upload/android/.build.gradle.swp b/node_modules/react-native-background-upload/android/.build.gradle.swp
new file mode 100644
index 0000000..64c3944
Binary files /dev/null and b/node_modules/react-native-background-upload/android/.build.gradle.swp differ
diff --git a/node_modules/react-native-background-upload/android/build.gradle b/node_modules/react-native-background-upload/android/build.gradle
index 9262df3..ba3e21c 100755
--- a/node_modules/react-native-background-upload/android/build.gradle
+++ b/node_modules/react-native-background-upload/android/build.gradle
@@ -19,7 +19,7 @@ android {
     buildToolsVersion project.hasProperty('buildToolsVersion') ? project.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
 
     defaultConfig {
-        minSdkVersion 18
+        minSdkVersion 16
         targetSdkVersion project.hasProperty('targetSdkVersion') ? project.targetSdkVersion : DEFAULT_TARGET_SDK_VERSION
         versionCode 1
         versionName "1.0"
