diff --git a/node_modules/react-native-keychain/index.js b/node_modules/react-native-keychain/index.js
index b73cfb2..d18ee61 100644
--- a/node_modules/react-native-keychain/index.js
+++ b/node_modules/react-native-keychain/index.js
@@ -94,10 +94,10 @@ type NormalizedOptions = {
   ...BaseOptions,
 };
 
-export type Options = {
+export type Options = $Shape<{|
   authenticationPrompt?: string | AuthenticationPrompt,
   ...BaseOptions,
-};
+|}>;
 
 export type Result = {|
   +service: string,
