diff --git a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
index d93debf..e22e234 100644
--- a/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
+++ b/node_modules/react-native-orientation-locker/iOS/RCTOrientation/Orientation.m
@@ -21,7 +21,7 @@ @implementation Orientation
 }
 
 #if (!TARGET_OS_TV)
-static UIInterfaceOrientationMask _orientationMask = UIInterfaceOrientationMaskAll;
+static UIInterfaceOrientationMask _orientationMask = UIInterfaceOrientationMaskPortrait;
 
 + (void)setOrientation: (UIInterfaceOrientationMask)orientationMask {
     _orientationMask = orientationMask;
