diff --git a/node_modules/remotedev/src/devTools.js b/node_modules/remotedev/src/devTools.js
index 743714b..a94ff2d 100644
--- a/node_modules/remotedev/src/devTools.js
+++ b/node_modules/remotedev/src/devTools.js
@@ -57,6 +57,9 @@ export function start(options) {
     }
   }
   connectToServer(options);
+  socket.on('error', e => {
+    // defining this handler suppresses errors if redux-devtools isn't open
+  });
 }
 
 function transformAction(action, config) {
