diff --git a/node_modules/stream-browserify/node_modules/readable-stream/readable.js b/node_modules/stream-browserify/node_modules/readable-stream/readable.js
index 2a8b5c6..ec73079 100644
--- a/node_modules/stream-browserify/node_modules/readable-stream/readable.js
+++ b/node_modules/stream-browserify/node_modules/readable-stream/readable.js
@@ -1,10 +1,10 @@
 exports = module.exports = require('./lib/_stream_readable.js');
-exports.Stream = require('stream');
+exports.Stream = require('stream-browserify');
 exports.Readable = exports;
 exports.Writable = require('./lib/_stream_writable.js');
 exports.Duplex = require('./lib/_stream_duplex.js');
 exports.Transform = require('./lib/_stream_transform.js');
 exports.PassThrough = require('./lib/_stream_passthrough.js');
 if (!process.browser && process.env.READABLE_STREAM === 'disable') {
-  module.exports = require('stream');
+  module.exports = require('stream-browserify');
 }
